zerver/lib/presence.py

```diff
@@ -147,14 +147,25 @@ def get_presence_for_user(
 
 
 def get_presence_dict_by_realm(
-    realm: Realm, slim_presence: bool = False, requesting_user_profile: Optional[UserProfile] = None
-) -> Dict[str, Dict[str, Any]]:
+    realm: Realm,
+    slim_presence: bool = False,
+    last_update_id_fetched_by_client: Optional[int] = None,
+    requesting_user_profile: Optional[UserProfile] = None,
+) -> Tuple[Dict[str, Dict[str, Any]], Optional[int]]:
     two_weeks_ago = timezone_now() - timedelta(weeks=2)
+    kwargs: Dict[str, object] = dict()
+    if last_update_id_fetched_by_client is not None:
+        kwargs["last_update_id__gt"] = last_update_id_fetched_by_client
+    else:
+        # TODO: How should clients set last_update_id_fetched_by_client at initialization?
+        # Should they do last_update_id_fetched_by_client = -1 to get *ALL* data?
+        kwargs["last_connected_time__gte"] = two_weeks_ago
```

Comment:
Yeah, I think `-1` is probably correct, if the client . Worth mentioning this detail in "api design", since we'll want to document it in the API docs for the main presence API endpoint.

As a side note, we should probably aim to delete or rework `docs/subsystems/presence.md` to mainly link to the API docs with some extra notes, following the completion of this project and documenting the modern presence API. But probably not worth getting distracted with that until we can merge this.

Reply:
https://chat.zulip.org/#narrow/stream/378-api-design/topic/presence.20rewrite.20.3E.20how.20much.20data.20to.20fetch.20at.20initialization

Reply:
Removed this block to have `last_connected_time__gte` inside the main query unconditionally with a note:
```
        # We can consider tweaking this value when last_update_id is being used,
        # to potentially fetch more data since such a client is expected to only
        # do it once and then only do small, incremental fetches.
        last_connected_time__gte=two_weeks_ago,
```

Reply:
Looks good!